kore/src/Kore/Rewrite/Axiom/Matcher.hs

```diff
@@ -271,6 +271,8 @@ matchEqualHeads (Pair (DV_ sort1 dv1) (DV_ sort2 dv2)) = do
     push (Pair dv1 dv2)
 matchEqualHeads (Pair (Equals_ _ _ term11 term12) (Equals_ _ _ term21 term22)) =
     push (Pair term11 term21) >> push (Pair term12 term22)
+matchEqualHeads (Pair (Not_ _ term1) (Not_ _ term2)) =
+    push (Pair term1 term2)
```

Comment:
Turned out this was what made the tests fail - the matcher was not ready for a `Not`.
We could complete more missing cases here but since a rewrite is waiting around the corner I did not.